packages/web3-eth-accounts/src/account.ts

```diff
@@ -53,10 +81,129 @@ export const privateKeyToAccount = (
 	const publicKeyString = `0x${publicKey.slice(2)}`;
 	const publicHash = sha3Raw(publicKeyString);
 	const publicHashHex = bytesToHex(publicHash);
-	const address = toChecksumAddress(publicHashHex.slice(-40)); // To get the address, take the last 20 bytes of the public hash
-	return { address, privateKey: stringPrivateKey, signTransaction, sign, encrypt };
+	return toChecksumAddress(publicHashHex.slice(-40)); // To get the address, take the last 20 bytes of the public hash
 };
 
+/**
+ * encrypt a private key given a password, returns a V3 JSON Keystore
+ * https://github.com/ethereum/wiki/wiki/Web3-Secret-Storage-Definition
+ */
+export const encrypt = async (
+	privateKey: HexString,
+	password: string | Buffer,
+	options?: CipherOptions,
+): Promise<Keystore> => {
+	if (!(isValidString(privateKey) || isBuffer(privateKey))) {
+		throw new InvalidPrivateKeyError(privateKey);
+	}
+
+	// if given salt or iv is a string, convert it to a Uint8Array
+	let salt;
+	if (options?.salt) {
+		salt = typeof options.salt === 'string' ? Buffer.from(options.salt, 'hex') : options.salt;
+	} else {
+		salt = randomBytes(32);
+	}
+	validateBytesInput(salt);
+
+	let iv;
```

Comment:
Could `iv` get a more descriptive name, or is that what it's expected to be called?